addons/stock_landed_costs/tests/test_stock_landed_costs_purchase.py

```diff
@@ -0,0 +1,246 @@
+            'picking_id': self.picking_in.id,
+            'location_id': self.supplier_location_id,
+            'location_dest_id': self.stock_location_id})
+        # Create picking outgoing shipment
+        self.picking_out = self.Picking.create({
+            'partner_id': self.customer_id,
+            'picking_type_id': self.picking_type_out_id,
+            'location_id': self.stock_location_id,
+            'location_dest_id': self.customer_location_id})
+        self.Move.create({
+            'name': self.product_refrigerator.name,
+            'product_id': self.product_refrigerator.id,
+            'product_uom_qty': 2,
+            'product_uom': self.product_refrigerator.uom_id.id,
+            'picking_id': self.picking_out.id,
+            'location_id': self.stock_location_id,
+            'location_dest_id': self.customer_location_id})
+
+    def test_00_landed_costs_on_incoming_shipment(self):
+        """ Test landed cost on incoming shipment """
+        #
+        # (A) Purchase product
+
+        #         Services           Quantity       Weight      Volume
+        #         -----------------------------------------------------
+        #         1. Refrigerator         5            10          1
+        #         2. Oven                 10           20          1.5
+
+        # (B) Add some costs on purchase
+
+        #         Services           Amount     Split Method
+        #         -------------------------------------------
+        #         1.labour            10        By Equal
+        #         2.brokerage         150       By Quantity
+        #         3.transportation    250       By Weight
+        #         4.packaging         20        By Volume
+
+        # Process incoming shipment
+        income_ship = self._process_incoming_shipment()
+        # Create landed costs
+        stock_landed_cost = self._create_landed_costs({
+            'equal_price_unit': 10,
+            'quantity_price_unit': 150,
+            'weight_price_unit': 250,
+            'volume_price_unit': 20}, income_ship)
+        # Compute landed costs
+        stock_landed_cost.compute_landed_cost()
+
+        valid_vals = {
+            'equal': 5.0,
+            'by_quantity_refrigerator': 50.0,
+            'by_quantity_oven': 100.0,
+            'by_weight_refrigerator': 50.0,
+            'by_weight_oven': 200,
+            'by_volume_refrigerator': 5.0,
+            'by_volume_oven': 15.0}
+
+        # Check valuation adjustment line recognized or not
+        self._validate_additional_landed_cost_lines(stock_landed_cost, valid_vals)
+        # Validate the landed cost.
+        stock_landed_cost.button_validate()
+        self.assertTrue(stock_landed_cost.account_move_id, 'Landed costs should be available account move lines')
+        account_entry = self.env['account.move.line'].read_group(
+            [('move_id', '=', stock_landed_cost.account_move_id.id)], ['debit', 'credit', 'move_id'], ['move_id'])[0]
```

Comment:
Checking that debit and credit are equal, does not test if the amounts are correct.